File: algorithms/1507.reformat-date/solution.rs
impl Solution {
    pub fn reformat_date(date: String) -> String {
        let mut date = date.split_whitespace().collect::<Vec<&str>>();

        let months = vec!["Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"];
        let month = format!("{:02}", (months.iter().position(|&m| m == date[1]).unwrap() + 1));

        let day = format!("{:02}", date[0].replace(&['d', 'h', 'n', 'r', 's', 't'][..], "").parse::<u8>().unwrap());

        [date[2], &month, &day].join("-")
    }
}

File: algorithms/1507.reformat-date/solution.rb
# @param {String} date
# @return {String}
def reformat_date(date)
  ## use Date ##
  # require 'date'
  # Date.parse(date).to_s

  year, month, day = date.split(' ').reverse

  months = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec']
  month = (months.find_index { |m| m == month } + 1).to_s.rjust(2, '0')

  # remove st, nd, rd, th
  day = day.tr('dhnrst', '').rjust(2, '0')

  [year, month, day].join('-')
end
